shell/common/promise_util.h

```diff
@@ -162,23 +163,36 @@ class Promise {
                                mate::ConvertToV8(isolate(), value));
   }
 
-  template <typename T>
-  v8::Maybe<bool> Reject(const T& value) {
+  v8::Maybe<bool> ResolveWithGin(const RT& value) {
+    static_assert(!std::is_same<void*, RT>(),
+                  "void* promises can not be resolved with a value");
     v8::HandleScope handle_scope(isolate());
     v8::MicrotasksScope script_scope(isolate(),
                                      v8::MicrotasksScope::kRunMicrotasks);
     v8::Context::Scope context_scope(
         v8::Local<v8::Context>::New(isolate(), GetContext()));
 
-    return GetInner()->Reject(GetContext(),
-                              mate::ConvertToV8(isolate(), value));
+    return GetInner()->Resolve(GetContext(),
+                               gin::ConvertToV8(isolate(), value));
   }
 
-  v8::Maybe<bool> RejectWithErrorMessage(const std::string& error);
+  v8::Maybe<bool> RejectWithErrorMessage(const std::string& string) {
+    v8::HandleScope handle_scope(isolate());
+    v8::MicrotasksScope script_scope(isolate(),
+                                     v8::MicrotasksScope::kRunMicrotasks);
+    v8::Context::Scope context_scope(
+        v8::Local<v8::Context>::New(isolate(), GetContext()));
+
+    v8::Local<v8::String> error_message =
+        v8::String::NewFromUtf8(isolate(), string.c_str(),
+                                v8::NewStringType::kNormal,
+                                static_cast<int>(string.size()))
+            .ToLocalChecked();
+    v8::Local<v8::Value> error = v8::Exception::Error(error_message);
+    return GetInner()->Reject(GetContext(), (error));
```

Comment:
_(nit)_ the parens around error don't seem to do anything?

Reply:
I just copied this over from the impl, you're right though :D will update